metrics/web/lib/common/data/repositories/firestore_instant_config_repository.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:metrics/common/data/model/instant_config_data.dart';
+import 'package:metrics/common/domain/entities/instant_config.dart';
+import 'package:metrics/common/domain/entities/persistent_store_error_code.dart';
+import 'package:metrics/common/domain/entities/persistent_store_exception.dart';
+import 'package:metrics/common/domain/repositories/instant_config_repository.dart';
+
+/// An implementation of the [InstantConfigRepository] for [Firestore].
+class FirestoreInstantConfigRepository extends InstantConfigRepository {
+  final Firestore _firestore = Firestore.instance;
+
+  @override
+  Future<InstantConfig> fetch() {
+    return _firestore
```

Comment:
Let's rewrite it using async-await and try-catch block to make it more readable and clear.